sylvia-derive/src/interfaces.rs

```diff
@@ -18,33 +18,40 @@ impl Interfaces {
         Self { interfaces }
     }
 
-    pub fn emit_glue_message_variants(&self, msg_ty: &MsgType) -> Vec<TokenStream> {
-        let sylvia = crate_module();
-
+    pub fn emit_glue_message_variants(
+        &self,
+        msg_ty: &MsgType,
+        contract: &Type,
+    ) -> Vec<TokenStream> {
         self.interfaces
             .iter()
             .map(|interface| {
                 let ContractMessageAttr {
-                    module,
-                    variant,
-                    generics,
-                    ..
+                    module, variant, ..
                 } = interface;
 
-                let generics = if !generics.is_empty() {
-                    quote! { < #generics > }
-                } else {
-                    quote! {}
-                };
-                let interface_enum =
-                    quote! { <#module ::sv::Api #generics as #sylvia ::types::InterfaceApi> };
+                let interface_enum = quote! { < #contract as #module ::sv::InterfaceMessagesApi> };
                 let type_name = msg_ty.as_accessor_name();
 
                 quote! { #variant ( #interface_enum :: #type_name) }
             })
             .collect()
     }
 
+    pub fn emit_glue_message_types(&self, msg_ty: &MsgType, contract: &Type) -> Vec<TokenStream> {
+        self.interfaces
+            .iter()
+            .map(|interface| {
+                let ContractMessageAttr { module, .. } = interface;
+
+                let interface_enum = quote! { < #contract as #module ::sv::InterfaceMessagesApi> };
```

Comment:
Ideally I would remove the `InterfaceMessagesApi` and change `InterfaceApi` to be generic over the `Contract: Interface`. But for this change we may go with this solution as a bridge.

Reply:
Should be solved by https://github.com/CosmWasm/sylvia/issues/363